lib/rucio/core/rule.py

```diff
@@ -1317,6 +1317,68 @@ def reduce_rule(rule_id, copies, exclude_expression=None, session=None):
         raise RuleNotFound('No rule with the id %s found' % (rule_id))
 
 
+@transactional_session
+def move_rule(rule_id, rse_expression, session=None):
+    """
+    Move a replication rule to another RSE and, once done, delete the original one.
+
+    :param rule_id:             Rule to be moved.
+    :param rse_expression:      RSE expression of the new rule.
+    :param session:             The DB Session.
+    :raises:                    RuleNotFound, RuleReplaceFailed
+    """
+    try:
+        rule = session.query(models.ReplicationRule).filter_by(id=rule_id).one()
+
+        if rule.state != RuleState.OK:
+            raise RuleReplaceFailed('The source rule must be in state OK.')
+
+        if rule.grouping == RuleGrouping.ALL:
+            grouping = 'ALL'
+        elif rule.grouping == RuleGrouping.NONE:
+            grouping = 'NONE'
+        else:
+            grouping = 'DATASET'
+
```

Comment:
Some pythonic way of doing this: 
grouping =  {RuleGrouping.ALL: 'ALL', RuleGrouping.NONE: 'NONE'}.get(rule.grouping, 'DATASET')

less readable though